wagtail/core/models.py

```diff
@@ -122,7 +123,7 @@ def find_for_request(request):
 
     @staticmethod
     def _find_for_request(request):
-        hostname = request.get_host().split(':')[0]
+        hostname = split_domain_port(request.get_host())[0]
         port = request.get_port()
```

Comment:
why not do `hostname, port = split_domain_port(request.get_host())` ?

Reply:
Because the implementation if `request.get_port` and `split_domain_port` are different.

Get port uses the request meta headers.
```
    def get_port(self):
        """Return the port number for the request as a string."""
        if settings.USE_X_FORWARDED_PORT and 'HTTP_X_FORWARDED_PORT' in self.META:
            port = self.META['HTTP_X_FORWARDED_PORT']
        else:
            port = self.META['SERVER_PORT']
        return str(port)
```
Split domain port does something different:
```
def split_domain_port(host):
    """
    Return a (domain, port) tuple from a given host.

    Returned domain is lowercased. If the host is invalid, the domain will be
    empty.
    """
    host = host.lower()

    if not host_validation_re.match(host):
        return '', ''

    if host[-1] == ']':
        # It's an IPv6 address without a port.
        return host, ''
    bits = host.rsplit(':', 1)
    domain, port = bits if len(bits) == 2 else (bits[0], '')
    # Remove a trailing dot (if present) from the domain.
    domain = domain[:-1] if domain.endswith('.') else domain
    return domain, port
```
So I do not dare to touch it.

Reply:
👍

Reply:
TIL: `request.get_host` uses `split_domain_port` under the hood - https://github.com/django/django/blob/e348ab0d4382b0d7cb0cab9d1261c916c3d0ce6c/django/http/request.py#L128

All good then!